.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,51 @@
+name: Publish docker image
```

Comment:
let's clarify what this workflow is in the file name and the title here, e.g. `Publish docker image for push to main branch' and the file name could be like `build-and-push-main.yml`?